server/Dockerfile

```diff
@@ -19,29 +22,36 @@ RUN apt-get update -y && \
     gcc build-essential binutils cmake extra-cmake-modules libsqlite3-mod-spatialite && \
     rm -rf /var/lib/apt/lists/*
 
-
 # needed for geodiff
 RUN pip3 install --upgrade pip
 RUN ln -s /usr/lib/x86_64-linux-musl/libc.so /lib/libc.musl-x86_64.so.1
 
-# create mergin user to run container with
-RUN groupadd -r mergin -g 901
-RUN groupadd -r mergin-family -g 999
-RUN useradd -u 901 -r --home-dir /app --create-home -g mergin -G mergin-family -s /sbin/nologin  mergin
-
-# copy app files
-COPY . /app
-WORKDIR /app
+COPY ./Pipfile.lock /app
+COPY ./Pipfile /app
 
 RUN pip3 install pipenv==2022.7.24
 # for locale check this http://click.pocoo.org/5/python3/
 ENV LC_ALL=C.UTF-8
 ENV LANG=C.UTF-8
 
 RUN pipenv install --system --deploy --verbose
-RUN pip3 install flower==0.9.7
+RUN pip3 install celery==5.2.2 kombu==5.2.2
 
+ARG UID=901
+ARG GID=901
+
+# create mergin user to run container with
+RUN groupadd -r mergin -g "${GID}"
+RUN groupadd -r mergin-family -g 999
+RUN useradd -u "${UID}" --home-dir /home/mergin/app -r -g mergin -G mergin-family -s /sbin/nologin  mergin
+
+COPY --chown=mergin:mergin . /home/mergin/app
+RUN chmod -R 755 /home/mergin/app
+
+# Set the non-root user as the default user
 USER mergin
 
-COPY ./entrypoint.sh .
-ENTRYPOINT ["./entrypoint.sh"]
+# Set working directory for non root user
+WORKDIR /home/mergin/app
```

Comment:
I'm not sure for this change. We can leave it in previous directory ```/app```.

Reply:
there were some problems with ownership but after review with @domfre we will try to revert it accordingly

Reply:
The point is, on the very top you set the WORKDIR to /app, however if we want to make the container to be runnable as non-root, we cannot set the home directory of the non-root user to be the same as the top workdir, otherwise we run into permission errors. So either we rename the top dir to something else or we rename the home directory of the mergin user (i.e. the non-root user).